File: python/bottle-song/bottle_song.py
def recite(start, take=1):
    def number_to_words(n):
        words = {
            0: "no",
            1: "One",
            2: "Two",
            3: "Three",
            4: "Four",
            5: "Five",
            6: "Six",
            7: "Seven",
            8: "Eight",
            9: "Nine",
            10: "Ten"
        }
        return words.get(n, str(n))

    def number_to_words_lower(n):
        word = number_to_words(n)
        return word.lower() if n != 0 else word

    verses = []
    for n in range(start, start - take, -1):
        current = n
        next_bottle = n - 1

        # Determine current bottle wording
        if current == 1:
            current_bottles = "One green bottle"
        else:
            current_bottles = f"{number_to_words(current)} green bottles"

        # Determine next bottle wording
        if next_bottle == 1:
            next_bottles = "one green bottle"
        elif next_bottle == 0:
            next_bottles = "no green bottles"
        else:
            next_bottles = f"{number_to_words_lower(next_bottle)} green bottles"

        verses.append(f"{current_bottles} hanging on the wall,")
        verses.append(f"{current_bottles} hanging on the wall,")
        verses.append("And if one green bottle should accidentally fall,")
        verses.append(f"There'll be {next_bottles} hanging on the wall.")

        if n != start - take + 1:
            verses.append("")

    return verses

File: java/bottle-song/src/main/java/BottleSong.java
class BottleSong {

    String recite(int startBottles, int takeDown) {
        StringBuilder song = new StringBuilder();
        for (int i = 0; i < takeDown; i++) {
            int current = startBottles - i;
            int next = current - 1;

            String currentWord = numberToWords(current, true);
            String nextWord = numberToWords(next, false);

            String currentBottle = current == 1 ? "bottle" : "bottles";
            String nextBottle = next == 1 ? "bottle" : "bottles";

            song.append(currentWord).append(" green ").append(currentBottle).append(" hanging on the wall,\n");
            song.append(currentWord).append(" green ").append(currentBottle).append(" hanging on the wall,\n");
            song.append("And if one green bottle should accidentally fall,\n");
            song.append("There'll be ").append(nextWord).append(" green ").append(nextBottle).append(" hanging on the wall.\n");

            if (i < takeDown - 1) {
                song.append("\n");
            }
        }
        return song.toString();
    }

    private String numberToWords(int number, boolean capitalize) {
        String word;
        switch (number) {
            case 10: word = "ten"; break;
            case 9: word = "nine"; break;
            case 8: word = "eight"; break;
            case 7: word = "seven"; break;
            case 6: word = "six"; break;
            case 5: word = "five"; break;
            case 4: word = "four"; break;
            case 3: word = "three"; break;
            case 2: word = "two"; break;
            case 1: word = "one"; break;
            case 0: word = "no"; break;
            default: word = Integer.toString(number); break;
        }
        if (capitalize && word.length() > 0) {
            word = word.substring(0,1).toUpperCase() + word.substring(1);
        }
        return word;
    }

}